src/awkward/cpp-headers/awkward/GrowableBuffer.h

```diff
@@ -142,6 +154,37 @@ namespace awkward {
       }
     }
 
+    template <typename TO_PRIMITIVE>
+    typename std::enable_if<!awkward::is_tt<std::complex, TO_PRIMITIVE>::value &&
+                             awkward::is_tt<std::complex, PRIMITIVE>::value>::type
+    copy_as(TO_PRIMITIVE* to_ptr, size_t offset) {
+      for (size_t i = 0; i < length_; i++) {
+        to_ptr[offset++] = static_cast<TO_PRIMITIVE>(ptr_.get()[i].real());
+        to_ptr[offset++] = static_cast<TO_PRIMITIVE>(ptr_.get()[i].imag());
+      }
```

Comment:
This template specialization expands a complex type array into an arithmetic type array by converting  the real and imaginary components into the arithmetic types. The length of a new buffer is double of the original.

Reply:
Oh, I see: because `offset` is incremented _twice_ in this loop over `length_`.